CompactedSyntax/CompactedPDFSyntaxTest.pdf.json

```diff
@@ -0,0 +1,1658 @@
+[
+  {
+    "IndirectObjectNumber": 1,
+    "IndirectObject": {
+      "Type": "Dictionary",
+      "Value": [
+        {
+          "Key": "Type",
+          "Value": {
+            "Type": "Name",
+            "Value": "Catalog"
+          }
+        },
+        {
+          "Key": "MarkInfo",
+          "Value": {
+            "Type": "Dictionary",
+            "Value": [
+              {
+                "Key": "Marked",
+                "Value": {
+                  "Type": "Boolean",
+                  "Value": true
+                }
+              },
+              {
+                "Key": "Suspects",
+                "Value": {
+                  "Type": "Boolean",
+                  "Value": true
+                }
+              },
+              {
+                "Key": "UserProperties",
+                "Value": {
+                  "Type": "Boolean",
+                  "Value": true
+                }
+              }
+            ]
+          }
+        },
+        {
+          "Key": "StructTreeRoot",
+          "Value": {
+            "Type": "Null"
+          }
+        },
+        {
+          "Key": "AA",
+          "Value": {
+            "Type": "Dictionary",
+            "Value": [
+              {
+                "Key": "WP",
+                "Value": {
+                  "Type": "Dictionary",
+                  "Value": [
+                    {
+                      "Key": "S",
+                      "Value": {
+                        "Type": "Name",
+                        "Value": "JavaScript"
+                      }
+                    },
+                    {
+                      "Key": "JS",
+                      "Value": {
+                        "Type": "LiteralString",
+                        "Value": "//JavaScript comment\napp.alert( \"Document Will Print Action!!\")"
```

Comment:
The PDF file has 0x0D 0x0A as the bytes in the literal string between the JS comment and the app.alert. It would be better to represent this precisely.

Reply:
The 1.7 PDF specification states the following syntax rule:

> An end-of-line marker appearing within a literal string without a preceding REVERSE SOLIDUS shall be treated as a byte value of (0Ah), irrespective of whether the end-of-line marker was a CARRIAGE RETURN (0Dh), a LINE FEED (0Ah), or *both*.

Given that the above appears in the Syntax chapter, I would consider the above to be in the scope of the lexical analyser. Syntax paragraph 7.2.1 explicitly defines the syntax rules to group bytes into token.

Reply:
OK - I agree with that. No change necessary. Thanks.